### ISSUE TITLE
Cookie file not created

### ISSUE THREAD

**Issue by Kixunil** [apotdevin/thunderhub#107]
**Describe the problem/bug**
When using `COOKIE_PATH`, the cookie is not created.

**Your environment**
* Version of ThunderHub: 0.8.13
* Deployment method: Debian package
* Other relevant environment details: Debian 10, nodejs v10.21.0

**To Reproduce**
1. Set `COOKIE_PATH` to an existing directory which allows thunderhub user to access it (see below)
2. Start thunderhub
3. `ls $directory`
4. See cookie file absent

**Expected behavior**
Cookie file should be present

**Actual behavior**
Cookie file is not created, no message in logs. The site can be accessed but it says "An unexpected error has occurred."

**Additional context**

env file:

```bash
BIND_PORT=4002

COOKIE_PATH=/var/lib/thunderhub-system-regtest/sso/cookie

DISABLE_LINKS=true

FETCH_FEES=false

FETCH_PRICES=false

HODL_HODL=false

NO_CLIENT_ACCOUNTS=true

NO_VERSION_CHECK=true

SSO_CERT_PATH=/var/lib/lnd-system-regtest/public/tls.cert

SSO_MACAROON_PATH=/var/lib/lnd-system-regtest/private/admin.macaroon

SSO_SERVER_URL=127.0.0.1:10011
```

Log:

```
Jul 29 16:06:53 disp7975 systemd[1]: Starting Lightning Node Manager...
Jul 29 16:06:53 disp7975 systemd[1]: Started Lightning Node Manager.
Jul 29 16:06:56 disp7975 thunderhub[5014]: ready - started server on http://localhost:4002
```

**Comment by apotdevin** [comment]
If there is no cookie it isn't created until the user visits at least one time the thunderhub page. This is due to the graphql server being serverless so it doesnt run until it gets a request (And then if it sees it has no cookie then it creates it).

To solve this, for example for the BTCPay integration, theres a small script that creates the cookie before starting thunderhub

https://github.com/apotdevin/thunderhub/blob/master/scripts/initCookie.sh

Would something like this work for you?

**Comment by Kixunil** [comment]
Hmm, what a weird chicken-egg problem. :D OK, I will try to create it and see if it works.

Why is there a fixed value in the script? It looks very dangerous. Why not `head -c 18 /dev/urandom | base64 > $FILE`?

**Comment by apotdevin** [comment]
Yeah, it could be done automatically by creating a custom server for nextjs (since it uses it's own) but it's too much hassle for a simple thing that can be done with the script

The cookie gets regenerated after every use so thought it was highly unlikely that it could be used maliciously with the initial value. Unless of course someone uses the script and then never opens thunderhub

**Comment by apotdevin** [comment]
But could change it as well to be random just to be on the safe side :)

**Comment by Kixunil** [comment]
I added the generation to the script but I still get `An unexpected error has occurred.` Still no logs. Will try to bump debug level tomorrow just in case.

**Comment by Kixunil** [comment]
Oh, turned out I forgot to add rewrite rule to `nginx`. Now I have a separate problem, will open a new issue.
